File: 7KYU/Valid Parentheses/solution.cs
using System.Collections.Generic;
using System.Linq;
public class Kata
{
  public static bool ValidParentheses(string str)
  {
    Stack<byte> stack = new Stack<byte>();
    
    for(var i = 0; i < str.Length; i++)
    {
      if(str[i] == '(')
        stack.Push(0);
      else
      {
         if(stack.Any())
           stack.Pop();
         else
          return false;
      }
    }
    return !stack.Any();
  } 
}

File: 7KYU/Valid Parentheses/solution.cpp
#include <string>

bool validParentheses(const std::string& str) {
  int open = 0;
  int i = 0;
  while(str[i] != '\0') {
    if(str[i] == '(') {
      open++;
    }
    else {
      open--;
      if(open < 0)
        return false;
    }
    i++;
  }
  return open == 0;
}